frontend/scenes/JoinLobbyScene.js

```diff
@@ -103,33 +103,51 @@ export default class JoinLobbyScene extends Phaser.Scene {
       });
   }
 
-  joinLobby(lobbyCode, playerName) {
-    fetch(`${server.api()}/lobbies/${lobbyCode}/join`, {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify({ username: playerName }),
-    })
-      .then((response) => {
-        if (!response.ok) {
-          return response.json().then((data) => {
-            throw new Error(data.error || "Failed to join lobby");
-          });
-        }
-        return response.json();
-      })
-      .then((data) => {
+  joinLobby(gameId, playerName) {
+    const ws = new WebSocket(server.websocket());
+
+    ws.onerror = (error) => {
+      console.error("WebSocket error:", error);
+    };
+
+    ws.onopen = () => {
+      const message = {
+        method: "join",
+        player_id: null,
+        game_id: gameId,
+        payload: {
+          player_name: playerName,
+        },
+      };
+
+      ws.send(JSON.stringify(message));
+    };
+
+    ws.onmessage = (event) => {
```

Comment:
maybe I'm missing something, but shouldn't we check that the method is "join"? considering we also have the pong message coming now and then?

Reply:
aah I see the sequence of events now 👍